titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/KubeNotificationProcessor.java

```diff
@@ -290,8 +290,13 @@ static Task updateTaskStatus(PodWrapper podWrapper,
                 .build();
 
         Task fixedTask = fillInMissingStates(podWrapper, updatedTask);
-        Task taskWithPlacementData = JobManagerUtil.attachPlacementData(fixedTask, executorDetailsOpt);
-        Task taskWithNodeMetadata = node.map(n -> attachNodeMetadata(taskWithPlacementData, n)).orElse(taskWithPlacementData);
+        Task taskWithExecutorData;
+        if (executorDetailsOpt.isPresent()) {
```

Comment:
instead of doing this here, did you consider changing the logic in `KubeUtil.getTitusExecutorDetails(pod)`, and checking whether `pod.getStatus().getPodIP()` should be used instead of annotations there? Centralizing this logic in that method looks cleaner to me. As more metadata is added to the kubelet path, I think it will be easier to keep both code paths more consistent in there.

Reply:
I feel like I'm getting conflicting review here. I originally had this logic in `getTitusExecutorDetails`, but @tbak asked me to check if executorDetailsOpt was present first and do it this way (pretty sure if I'm interpreting his intent).

Reply:
My point was that `TitusExecutorDetails` is a legacy data structure that we want to get rid off.
I think it would be better to move the new function into `KubeUtil`, but keep it separate as it is now.